### PR TITLE
Manually fix lint warnings and use `no-var` where possible

### PR DESCRIPTION
**Pull request by wishdasher**
Now with only 117 problems (46 errors, 71 warnings)!

### CODE REVIEW

File: src/components/ClassSearch.vue [hunk 1]
```diff
@@ -16,7 +16,7 @@
           :rows-per-page-items="rowsPerPageItems"
           :pagination.sync="pagination"
           :no-data-text="'No results'"
-          :rows-per-page-text="'Results per page: '"
+          :rows-per-page-text="'Results per page:\xa0'"
```
Comment (ttpcodes) [inline src/components/ClassSearch.vue]: What happened here?

File: src/pages/MainPage.vue [hunk 2]
```diff
@@ -36,7 +36,7 @@
         :roads="roads"
         :active-road="activeRoad"
         :subjects="subjectsInfo"
-        @delete-road="$refs.authcomponent.deleteRoad($event)"
+        @devare-road="$refs.authcomponent.devareRoad($event)"
```
Comment (ttpcodes) [inline src/pages/MainPage.vue]: Is this supposed to be renamed?

File: src/pages/MainPage.vue [hunk 3]
```diff
@@ -58,7 +58,7 @@
         :just-loaded="justLoaded"
         :active-road="activeRoad"
         :conflict-info="conflictInfo"
-        @delete-road="deleteRoad"
+        @devare-road="devareRoad"
```
Comment (ttpcodes) [inline src/pages/MainPage.vue]: Is this supposed to be renamed?

File: src/pages/MainPage.vue [hunk 4]
```diff
@@ -85,7 +85,7 @@
             id="searchInputTF"
             slot="activator"
             v-model="searchInput"
-            autocomplete="false"
+            autocompvare="false"
```
Comment (ttpcodes) [inline src/pages/MainPage.vue]: Is this supposed to be renamed?

File: src/pages/MainPage.vue [hunk 5]
```diff
@@ -453,10 +453,10 @@ export default {
       if (this.activeRoad === oldid) {
         this.activeRoad = newid;
       }
-      Vue.delete(this.roads, oldid);
+      Vue.devare(this.roads, oldid);
```
Comment (ttpcodes) [inline src/pages/MainPage.vue]: Is this supposed to be renamed?

File: src/pages/MainPage.vue [hunk 6]
```diff
@@ -521,8 +521,8 @@ export default {
     changeActiveRoad: function (event) {
       this.activeRoad = event;
     },
-    deleteRoad: function (roadID) {
-      Vue.delete(this.roads, roadID);
+    devareRoad: function (roadID) {
```
Comment (ttpcodes) [inline src/pages/MainPage.vue]: There's a lot of this `delete` being replaced by `devare`. That probably should be fixed.

File: src/components/ClassSearch.vue (hunk 1 above)
Comment (wishdasher) [reply src/components/ClassSearch.vue]: Originally, I copy+pasted a non-standard whitespace character. `\xa0` is a non-breaking space, which is a better way of doing this.

File: src/pages/MainPage.vue (hunk 2 above)
Comment (wishdasher) [reply src/pages/MainPage.vue]: Nope, result of bad find+replace. Thanks for catching it!

File: src/components/ClassSearch.vue (hunk 1 above)
Comment (wishdasher) [reply src/components/ClassSearch.vue]: More context: `eslint` didn't like the string having extra whitespace at the end.

**Comment by npfoss** [comment]
I'm getting a ton of console errors just from logging in and loading roads:
image
but I think I have figured out the problem and will push the solution shortly

**Pull request merged**